src/node/internal/internal_net.ts

```diff
@@ -1303,7 +1325,7 @@ function addClientAbortSignalOption(
   }
 
   if (signal.aborted) {
-    queueMicrotask(onAbort);
+    onAbort();
```

Comment:
This might be a limitation in Node.js also but... there is `signal.reason` which provides access to the error (or other value) that the `AbortSignal` was aborted with. If the `signal` is aborted then we really ought to be propagating that `reason`